scene/main/node.cpp

```diff
@@ -2104,12 +2105,22 @@ Node *Node::_duplicate(int p_flags) const {
 
 	get_property_list(&plist);
 
+	StringName script_property_name = CoreStringNames::get_singleton()->_script;
+
+	if (p_flags & DUPLICATE_SCRIPTS) {
+		bool is_valid = false;
+		Variant script = get(script_property_name, &is_valid);
+		if (is_valid) {
```

Comment:
Does this ever get called if it is set false within this block?

Reply:
`get(script_property_name, &is_valid);` is assigning it. You can look at `Object::get` implementation to see how it works.

Reply:
@Sslaxx notice the `&is_valid` as a parameter to `get()`, it indeed *might* get modified!